### ISSUE TITLE
GTK Initializing failed - can't start

### ISSUE THREAD

**Issue by peterandrewd** [sickcodes/Docker-OSX#43]
So I created a docker image using the guide. 
I run the command: docker run --privileged -v /tmp/.X11-unix:/tmp/.X11-unix docker-osx:latest
It recognises docker-osx:latest. It does its usual start up command sequence, then gets to the end and says: gtk initialization failed.
IF I start docker-osx normally without that tag, using the: docker run --privileged -e "DISPLAY=${DISPLAY:-:0.0}" -v /tmp/.X11-unix:/tmp/.X11-unix sickcodes/docker-osx command, it runs. But I want usb support so yeah.

**Comment by sickcodes** [comment]
I am quite sure I definitely can get iPhone full pass through to Xcode. Let me try today using virt manager.

**Comment by sickcodes** [comment]
You need to docker pull that latest version

`docker pull sickcodes/docker-osx`

**Comment by peterandrewd** [comment]
And then create the image?

**Comment by sickcodes** [comment]
`docker pull sickcodes/docker-osx

docker run --privileged -e "DISPLAY=${DISPLAY:-:0.0}" -v /tmp/.X11-unix:/tmp/.X11-unix sickcodes/docker-osx

`

Message me on Twitter if you want I’ll show you over teamviewer  😂

**Comment by sickcodes** [comment]
I’m not sure if you get it entirely but docker works on intermediate containers

When you `pull` down, you get a copy that has been built by hub.docker.com

It’s stored “in state”. Docker lets you run the second command in the comment above 10 times in a row each one in its own container.

The other option is to build it yourself.

**Comment by peterandrewd** [comment]
Yes that's what Im trying to do, a custom build with more RAM and USB support for iPhone.
Here is my command, is it correct?
docker run --privileged -e RAM=16 -e EXTRA='-usb -device usb-host,hostbus=1,hostaddr=3' -e "DISPLAY=${DISPLAY:-:0.0}" -v /tmp/.X11-unix:/tmp/.X11-unix sickcodes/docker-osx

**Comment by peterandrewd** [comment]
I mean using those commands it boots and installs Catalina...but progress bar gets stuck halfway on loading into the actual OS once it’s installed...

**Comment by sickcodes** [comment]
Check disk space usage, any screenshots?

**Comment by peterandrewd** [comment]
I have 300GB of free available disk space so it ain’t that.
Below in the photo it says libusb_set_configuration: -6 [BUSY] in the terminal.
That’s all it says.

![image](https://user-images.githubusercontent.com/66618946/84691384-677e5600-af87-11ea-96c0-3c8506128f4c.jpeg)

**Comment by peterandrewd** [comment]
docker run -e RAM=8 -e EXTRA='-usb -device usb-host,hostbus=1,hostaddr=2' -p 50922:10022 --privileged -e "DISPLAY=${DISPLAY:-:0.0}" -v /tmp/.X11-unix:/tmp/.X11-unix sickcodes/docker-osx:latest

That is the command I ran. I don't think on my end Im doing anything wrong this time. Everything loads and installs fine except when it actually comes time to boot into Catalina.
But when I don't put the -e EXTRA='-usb -device argument in, it boots in. Go figure.

**Comment by peterandrewd** [comment]
See here, because I created the original container using the argument "-e EXTRA='-usb -device usb-host,hostbus=1,hostaddr=2'", now qemu gives an error: "qemu-system-x86_64: -device usb-host,hostbus=1,hostaddr=2: failed to find host usb device 1:2" when using docker start xxxxxxxx, because the USB device was unplugged and replugged and thus changed device ID's.
So you cannot create a container using the original USB filter arguments.
Am I doing something wrong, or am I correct this time?

**Comment by sickcodes** [comment]
Yeah for one time use you can passthrough a device, it has to be in the same spot later. I think you’ll have better luck using the Virt-manager version outside of docker, you can add and remove usb devices willy nilly.

**Comment by peterandrewd** [comment]
> Yeah for one time use you can passthrough a device, it has to be in the same spot later. I think you’ll have better luck using the Virt-manager version outside of docker, you can add and remove usb devices willy nilly.

Ok and as I'm new and learning, how do I use the virt manager method?

**Comment by peterandrewd** [comment]
Nvm. The VM is installed in virt manager. Haven’t installed Catalina yet but how similar or how well it runs compared to the QEMU method though is another question...

**Comment by sickcodes** [comment]
It should be unnoticeably faster 😂

**Comment by peterandrewd** [comment]
Still same issue. Can't mount iPhone into "VM"

**Comment by laijinsheng** [comment]
Hello everyone, dear netizens, I'm a netizen from China. Because do not understand English, so the detailed operation will use Chinese to explain, please use translation software translation. Sorry for the inconvenience

------

大家好，各位亲爱的网友，我是来自中国的网友。因不懂英文 所以详细操作会使用中文阐述，请各大网友使用翻译软件翻译。不便之处请见谅

经过一下午的折腾终于看到了docker-osx界面

执行启动命令前先查询vnc端口是否被占有
`sudo netstat -tunlp | grep 5903`
若被占用可进行更换（注意启动命令也需要做出修改）或者kill掉占用的进程

我使用的启动命令为：
`docker run -p 5903:5903 -e EXTRA='-vnc localhost:3' --privileged --net host --cap-add=ALL -v /tmp/.X11-unix:/tmp/.X11-unix -v /dev:/dev -v /lib/modules:/lib/modules sickcodes/docker-osx`

执行后等待3-5分钟 使用vnc viever来链接。
链接地址为：localhost:5903
![1](https://user-images.githubusercontent.com/34933847/85122892-e3afbc80-b259-11ea-984b-7f6ab7d71015.jpg)
![2](https://user-images.githubusercontent.com/34933847/85122904-e5798000-b259-11ea-8ed3-9f2444c057fe.jpg)
![3](https://user-images.githubusercontent.com/34933847/85122921-ea3e3400-b259-11ea-9803-ee9cbfce547b.jpg)


过程中遇到的问题及分析思路：

问题1：gtk initialization failed
分析：因linux可视化界面基于xorg，所以需要改掉qume的启动方式（个人理解） 然后通过增加 -e EXTRA='-vnc localhost:3' 参数实现vnc转发

问题2：Could not access KVM kernel module: Permission denied、qemu-system-x86_64: failed to initialize kvm: Permission denied
结论：/dev/kvm权限问题 执行命令 sudo chmod 666 /dev/kvm

----

有疑问欢迎追加提问，感谢大家

If you have any questions, welcome to ask additional questions. Thank you

**Comment by sickcodes** [comment]
> Hello everyone, dear netizens, I'm a netizen from China. Because do not understand English, so the detailed operation will use Chinese to explain, please use translation software translation. Sorry for the inconvenience
> 
> 
> 
> ------
> 
> 
> 
> 大家好，各位亲爱的网友，我是来自中国的网友。因不懂英文 所以详细操作会使用中文阐述，请各大网友使用翻译软件翻译。不便之处请见谅
> 
> 
> 
> 经过一下午的折腾终于看到了docker-osx界面
> 
> 
> 
> 执行docker启动命令前先检查vnc端口是否被占有
> 
> `sudo netstat -tunlp | grep 5903`
> 
> 若被占用可进行更换（注意启动命令也需要做出修改）或者kill掉占用的进程
> 
> 
> 
> 然后我执行以下命令启动：
> 
> `docker run -p 5903:5903 -e EXTRA='-vnc localhost:3' --privileged --net host --cap-add=ALL -v /tmp/.X11-unix:/tmp/.X11-unix -v /dev:/dev -v /lib/modules:/lib/modules sickcodes/docker-osx`
> 
> 
> 
> 执行后等待3-5分钟 使用vnc viever来链接。
> 
> 链接地址为：localhost:5903
> 
> ![1](https://user-images.githubusercontent.com/34933847/85122892-e3afbc80-b259-11ea-984b-7f6ab7d71015.jpg)
> 
> ![2](https://user-images.githubusercontent.com/34933847/85122904-e5798000-b259-11ea-8ed3-9f2444c057fe.jpg)
> 
> ![3](https://user-images.githubusercontent.com/34933847/85122921-ea3e3400-b259-11ea-9803-ee9cbfce547b.jpg)
> 
> 
> 
> 
> 
> 过程中遇到的问题及分析思路：
> 
> 
> 
> 问题1：gtk initialization failed
> 
> 分析：因linux可视化界面基于xorg，所以需要改掉qume的启动方式（个人理解） 然后通过增加 -e EXTRA='-vnc localhost:3' 参数实现vnc转发
> 
> 
> 
> 问题2：Could not access KVM kernel module: Permission denied、qemu-system-x86_64: failed to initialize kvm: Permission denied
> 
> 结论：/dev/kvm权限问题 执行命令 sudo chmod 666 /dev/kvm
> 
> 
> 
> ----
> 
> 
> 
> 有疑问欢迎追加提问，感谢大家
> 
> 
> 
> If you have any questions, welcome to ask additional questions. Thank you
> 
> 
> 
> 
> 
> 

You cannot use VNC and $DISPLAY at the same time because they are different monitors.

You can use the VNC version already here 


vnc-version/Dockerfile

**Comment by laijinsheng** [comment]
> > Hello everyone, dear netizens, I'm a netizen from China. Because do not understand English, so the detailed operation will use Chinese to explain, please use translation software translation. Sorry for the inconvenience
> > 
> > 大家好，各位亲爱的网友，我是来自中国的网友。因不懂英文 所以详细操作会使用中文阐述，请各大网友使用翻译软件翻译。不便之处请见谅
> > 经过一下午的折腾终于看到了docker-osx界面
> > 执行docker启动命令前先检查vnc端口是否被占有
> > `sudo netstat -tunlp | grep 5903`
> > 若被占用可进行更换（注意启动命令也需要做出修改）或者kill掉占用的进程
> > 然后我执行以下命令启动：
> > `docker run -p 5903:5903 -e EXTRA='-vnc localhost:3' --privileged --net host --cap-add=ALL -v /tmp/.X11-unix:/tmp/.X11-unix -v /dev:/dev -v /lib/modules:/lib/modules sickcodes/docker-osx`
> > 执行后等待3-5分钟 使用vnc viever来链接。
> > 链接地址为：localhost:5903
> > ![1](https://user-images.githubusercontent.com/34933847/85122892-e3afbc80-b259-11ea-984b-7f6ab7d71015.jpg)
> > ![2](https://user-images.githubusercontent.com/34933847/85122904-e5798000-b259-11ea-8ed3-9f2444c057fe.jpg)
> > ![3](https://user-images.githubusercontent.com/34933847/85122921-ea3e3400-b259-11ea-9803-ee9cbfce547b.jpg)
> > 过程中遇到的问题及分析思路：
> > 问题1：gtk initialization failed
> > 分析：因linux可视化界面基于xorg，所以需要改掉qume的启动方式（个人理解） 然后通过增加 -e EXTRA='-vnc localhost:3' 参数实现vnc转发
> > 问题2：Could not access KVM kernel module: Permission denied、qemu-system-x86_64: failed to initialize kvm: Permission denied
> > 结论：/dev/kvm权限问题 执行命令 sudo chmod 666 /dev/kvm
> > 
> > 有疑问欢迎追加提问，感谢大家
> > If you have any questions, welcome to ask additional questions. Thank you
> 
> You cannot use VNC and $DISPLAY at the same time because they are different monitors.
> 
> You can use the VNC version already here
> 
> vnc-version/Dockerfile

感谢作者的回复，看了上面的一些提示后 进行了二次操作，使用一下命令报错。提示为硬盘空间不足。待解决后继续尝试 感谢

`docker run --privileged -e "DISPLAY=${DISPLAY:-:0.0}" -v /tmp/.X11-unix:/tmp/.X11-unix sickcodes/docker-osx`

**Comment by sickcodes** [comment]
The command you mentioned is for X11 display connection

Here are the VNC rules:

```

# Build:
#
#       # write down the password at the end
#       docker build -t docker-osx-vnc .
# 
# Run:
#       
#       docker run -p 8888:5999 -p 50922:10022 -d --privileged docker-osx-vnc:latest
#
#
# Connect locally (safe):
#
#       VNC Host:     localhost:8888


```

**Comment by peterandrewd** [comment]
Why is this issue closed? Its still a problem when using a custom build.
Yes it builds, yes it recognized the docker image file, and it tries to boot in with the command docker run --privileged -v /tmp/.X11-unix:/tmp/.X11-unix docker-osx:latest but says GTK initialising failed.
AND I have tried docker build -t docker-osx-vnc . Then running docker run -p 8888:5999 -p 50922:10022 -d --privileged docker-osx-vnc:latest and it doesn't even boot in at all. Just blank terminal after that.

**Comment by peterandrewd** [comment]
??

**Comment by sickcodes** [comment]
Hi Peter. You can simply clone the repo and do the lines at the top of the commented section of the file 

`docker build ...
docker run ...
`
The commands are written in full at the top of the Dockerfile.


Note that whatever label you choose, you need to run that image

**Comment by peterandrewd** [comment]
> Hi Peter. You can simply clone the repo and do the lines at the top of the commented section of the file 
> 
> `docker build ...
> docker run ...
> `
> The commands are written in full at the top of the Dockerfile.
> 
> 
> Note that whatever label you choose, you need to run that image

Yes that is what I did: docker build -t docker-osx:latest —build arg SIZE=64G

Then: docker run --privileged -CORES=4 -e RAM=8 -e "DISPLAY=${DISPLAY:-:0.0}" -v /tmp/.X11-unix:/tmp/.X11-unix sickcodes/docker-osx:latest

It says port in use and doesn’t run

**Comment by sickcodes** [comment]
docker build -t **docker-osx:latest**
docker run .... **sickcodes/docker-osx:latest**

These are two different Docker images.

**sickcodes/docker-osx:latest** is from hub.docker.com/sickcodes/docker-osx.
**docker-osx:latest** does not exist, unless you build it, which you did.

Since you built it, use that in the docker run command instead of the other one.

**Comment by hoaivan** [comment]
this is how i solved GTK initilizing failed on Manjaro, just run below command before starting docker: 
xhost +

**Comment by farhandiwan** [comment]
I am  trying to  run the docker container  from a Centos 8 VM created on top of Virtualbox.
Getting the same GTK error.  I did run xhosts + command to allow the xhosts.

getting a warning too for CPU flags for EAX not availble.

Using the latest osx image from  docker repo.

**Comment by farhandiwan** [comment]
which distribution is this tested on . I can  test on it.  I have choice of OS if that resolves the problem.

**Comment by sickcodes** [comment]
> I am trying to run the docker container from a Centos 8 VM created on top of Virtualbox.
> Getting the same GTK error. I did run xhosts + command to allow the xhosts.
> 
> getting a warning too for CPU flags for EAX not availble.
> 
> Using the latest osx image from docker repo.

Does this show 0 or 1?
`egrep -c '(svm|vmx)' /proc/cpuinfo`

**Comment by farhandiwan** [comment]
I did check the cpu flags  i did not  find svm but.  vmx. was present in the cpu flags.

**Comment by farhandiwan** [comment]
VMX. is present in cpu flags.

**Comment by joe733** [comment]
Hi, I pulled the `big sur` docker image: `docker pull sickcodes/docker-osx:big-sur`. Then according the [`initial setup`](https://github.com/sickcodes/Docker-OSX#initial-setup) I install the dependencies. Rebooted.

**System Info**
```dart
~ via 🐍 v3.6.9 
➜ neofetch 
         eeeeeeeeeeeeeeeee            jovial@xpsl502x 
      eeeeeeeeeeeeeeeeeeeeeee         --------------- 
    eeeee  eeeeeeeeeeee   eeeee       OS: elementary OS 5.1.7 Hera x86_64 
  eeee   eeeee       eee     eeee     Host: Dell System XPS L502X 
 eeee   eeee          eee     eeee    Kernel: 5.4.0-70-generic 
eee    eee            eee       eee   Uptime: 2 hours, 41 mins 
eee   eee            eee        eee   Packages: 2080 
ee    eee           eeee       eeee   Shell: bash 4.4.20 
ee    eee         eeeee      eeeeee   Resolution: 1366x768 
ee    eee       eeeee      eeeee ee   DE: Pantheon 
eee   eeee   eeeeee      eeeee  eee   WM: Mutter(Gala) 
eee    eeeeeeeeee     eeeeee    eee   Terminal: io.elementary.t 
 eeeeeeeeeeeeeeeeeeeeeeee    eeeee    CPU: Intel i5-2410M (4) @ 2.900GHz 
  eeeeeeee eeeeeeeeeeee      eeee     GPU: NVIDIA GeForce GT 525M 
    eeeee                 eeeee       GPU: Intel 2nd Generation Core Processor Family 
      eeeeeee         eeeeeee         Memory: 2358MiB / 11874MiB 
         eeeeeeeeeeeeeeeee 
                                                              
```

**Misc**
```dart
~ via 🐍 v3.6.9 
➜ docker -v
Docker version 19.03.13, build cd8016b6bc

~ via 🐍 v3.6.9 
➜ groups jovial
jovial : jovial adm cdrom sudo dip plugdev lpadmin sambashare docker kvm libvirt

~ via 🐍 v3.6.9 
➜ kvm-ok 
INFO: /dev/kvm exists
KVM acceleration can be used

~ via 🐍 v3.6.9 
➜ kvm --version
QEMU emulator version 2.11.1(Debian 1:2.11+dfsg-1ubuntu7.36)
Copyright (c) 2003-2017 Fabrice Bellard and the QEMU Project developers

~ via 🐍 v3.6.9 
➜ egrep -c '(svm|vmx)' /proc/cpuinfo
4

~ via 🐍 v3.6.9 
➜ xhost 
access control enabled, only authorized clients can connect
SI:localuser:jovial

~ via 🐍 v3.6.9 
➜ echo $DISPLAY
:0

~ via 🐍 v3.6.9 
➜ cat /sys/module/kvm/parameters/ignore_msrs
Y

~ via 🐍 v3.6.9 
➜ tail /sys/module/kvm_intel/parameters/nested
Y
```

I think my system is capable of running MacOS docker images. But when I run the command, I'm 🤕.

```dart
~ via 🐍 v3.6.9 
➜ docker run -it --name bigsur --device /dev/kvm -p 50922:10022 -v /tmp/.X11-unix:/tmp/.X11-unix -e "DISPLAY=${DISPLAY:-:0.0}" sickcodes/docker-osx
touch: cannot touch '': No such file or directory
chown: cannot access '': No such file or directory
ssh-keygen: generating new host keys: RSA DSA ECDSA ED25519 
++ id -u
++ id -g
+ sudo chown 1000:1000 /dev/kvm
nohup: appending output to 'nohup.out'
++ id -u
++ id -g
+ sudo chown -R 1000:1000 /dev/snd
+ [[ 3 = max ]]
+ [[ 3 = half ]]
++ id -u
++ id -g
+ sudo chown -R 1000:1000 /dev/snd
+ exec qemu-system-x86_64 -m 3000 -cpu Penryn,vendor=GenuineIntel,+invtsc,vmware-cpuid-freq=on,+pcid,+ssse3,+sse4.2,+popcnt,+avx,+aes,+xsave,+xsaveopt,check -machine q35,accel=kvm:tcg -smp 4,cores=4 -usb -device usb-kbd -device usb-tablet -device 'isa-applesmc,osk=ourhardworkbythesewordsguardedpleasedontsteal(c)AppleComputerInc' -drive if=pflash,format=raw,readonly,file=/home/arch/OSX-KVM/OVMF_CODE.fd -drive if=pflash,format=raw,file=/home/arch/OSX-KVM/OVMF_VARS-1024x768.fd -smbios type=2 -audiodev alsa,id=hda -device ich9-intel-hda -device hda-duplex,audiodev=hda -device ich9-ahci,id=sata -drive id=OpenCoreBoot,if=none,snapshot=on,format=qcow2,file=/home/arch/OSX-KVM/OpenCore-Catalina/OpenCore.qcow2 -device ide-hd,bus=sata.2,drive=OpenCoreBoot -device ide-hd,bus=sata.3,drive=InstallMedia -drive id=InstallMedia,if=none,file=/home/arch/OSX-KVM/BaseSystem.img,format=qcow2 -drive id=MacHDD,if=none,file=/home/arch/OSX-KVM/mac_hdd_ng.img,format=qcow2 -device ide-hd,bus=sata.4,drive=MacHDD -netdev user,id=net0,hostfwd=tcp::10022-:22,hostfwd=tcp::5900-:5900, -device vmxnet3,netdev=net0,id=net0,mac=52:54:00:09:49:17 -monitor stdio -vga vmware
Unable to init server: Could not connect: Connection refused
QEMU 5.2.0 monitor - type 'help' for more information
(qemu) ALSA lib confmisc.c:767:(parse_card) cannot find card '0'
ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_card_driver returned error: No such file or directory
ALSA lib confmisc.c:392:(snd_func_concat) error evaluating strings
ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory
ALSA lib confmisc.c:1246:(snd_func_refer) error evaluating name
ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
ALSA lib conf.c:5233:(snd_config_expand) Evaluate error: No such file or directory
ALSA lib pcm.c:2660:(snd_pcm_open_noupdate) Unknown PCM default
alsa: Could not initialize DAC
alsa: Failed to open `default':
alsa: Reason: No such file or directory
ALSA lib confmisc.c:767:(parse_card) cannot find card '0'
ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_card_driver returned error: No such file or directory
ALSA lib confmisc.c:392:(snd_func_concat) error evaluating strings
ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory
ALSA lib confmisc.c:1246:(snd_func_refer) error evaluating name
ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
ALSA lib conf.c:5233:(snd_config_expand) Evaluate error: No such file or directory
ALSA lib pcm.c:2660:(snd_pcm_open_noupdate) Unknown PCM default
alsa: Could not initialize DAC
alsa: Failed to open `default':
alsa: Reason: No such file or directory
audio: Failed to create voice `dac'
ALSA lib confmisc.c:767:(parse_card) cannot find card '0'
ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_card_driver returned error: No such file or directory
ALSA lib confmisc.c:392:(snd_func_concat) error evaluating strings
ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory
ALSA lib confmisc.c:1246:(snd_func_refer) error evaluating name
ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
ALSA lib conf.c:5233:(snd_config_expand) Evaluate error: No such file or directory
ALSA lib pcm.c:2660:(snd_pcm_open_noupdate) Unknown PCM default
alsa: Could not initialize ADC
alsa: Failed to open `default':
alsa: Reason: No such file or directory
ALSA lib confmisc.c:767:(parse_card) cannot find card '0'
ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_card_driver returned error: No such file or directory
ALSA lib confmisc.c:392:(snd_func_concat) error evaluating strings
ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_concat returned error: No such file or directory
ALSA lib confmisc.c:1246:(snd_func_refer) error evaluating name
ALSA lib conf.c:4745:(_snd_config_evaluate) function snd_func_refer returned error: No such file or directory
ALSA lib conf.c:5233:(snd_config_expand) Evaluate error: No such file or directory
ALSA lib pcm.c:2660:(snd_pcm_open_noupdate) Unknown PCM default
alsa: Could not initialize ADC
alsa: Failed to open `default':
alsa: Reason: No such file or directory
audio: Failed to create voice `adc'
gtk initialization failed
```

Ignoring ALSA errors, the other two that stand out are: `Unable to init server: Could not connect: Connection refused` and `gtk initialization failed`. There are a lot of [issues raised](https://github.com/sickcodes/Docker-OSX/issues?q=is%3Aissue+is%3Aclosed+gtk) around these, but they didn't solve mine.  I might me missing some small detail here can anybody help me, please?

**Comment by farhandiwan** [comment]
Having the same issue.
As GTK initialization failed.

**Comment by sickcodes** [comment]
```bash
chmod 666 /dev/kvm
```
Or
```bash
xhost +
```
Or 

```bash
sudo systemctl stop docker
sudo dockerd
docker run....
```

**Comment by joe733** [comment]
```dart
~ via 🐍 v3.6.9 ⏱️  5s
➜ sudo chmod 666 /dev/kvm
[sudo] password for jovial:       

~ via 🐍 v3.6.9 
➜ ls -l /dev/kvm
crw-rw-rw- 1 root kvm 10, 232 Apr 12 14:14 /dev/kvm
```

Nope that didn't work. Same error log.

``` dart
~ via 🐍 v3.6.9 
➜ xhost
access control disabled, clients can connect from any host
SI:localuser:jovial
```

As this https://github.com/sickcodes/Docker-OSX/issues/213#issuecomment-805000950 says `xhost +` isn't quite recommended. So earlier I had done `xhost +SI:localuser:username`, nevertheless I tried again with `xhost +` but it didn't work. Same error message.

Oh forgive me, I should have mention this earlier. [Docker is snapped](https://snapcraft.io/docker) on my system.
```dart
~ via 🐍 v3.6.9 
➜ snap list
Name       Version        Rev    Tracking       Publisher      Notes
...
docker     19.03.13       796    latest/stable  canonical*     -
...
```

Which is why I guess I do not have `docker.service` nor `dockerd`

```dart
~ via 🐍 v3.6.9 
➜ sudo systemctl stop docker
Failed to stop docker.service: Unit docker.service not loaded.

~ via 🐍 v3.6.9 
➜ sudo systemctl enable docker.service
Failed to enable unit: Unit file docker.service does not exist.

~ via 🐍 v3.6.9 
➜ dockerd --help

Command 'dockerd' not found, but can be installed with:

sudo apt install docker.io
```

I really do not want to use `apt` or `ppa` to install docker, it would mess up my other configuration. @sickcodes does this issue have something to do with snapped docker, is there anything that I can try on snap?

**Comment by BackMountainDevil** [comment]
I guess we have the same problem. But I have installed docker.

**Comment by fluxt** [comment]
@sickcodes I had the same exact error as @joe733, but I noticed that

> Oh forgive me, I should have mention this earlier. [Docker is snapped](https://snapcraft.io/docker) on my system.

So did I when I had the error. So, I uninstalled the [snap docker](https://snapcraft.io/docker) and installed with the [PPA one](https://docs.docker.com/engine/install/ubuntu/#install-using-the-repository), and now it works fine. Might be worth mentioning in the documentation that using docker snap doesn't work.

**Comment by sickcodes** [comment]
> @sickcodes I had the same exact error as @joe733, but I noticed that
> 
> 
> 
> > Oh forgive me, I should have mention this earlier. [Docker is snapped](https://snapcraft.io/docker) on my system.
> 
> 
> 
> So did I when I had the error. So, I uninstalled the snap docker and installed with the [PPA one](https://docs.docker.com/engine/install/ubuntu/#install-using-the-repository), and now it works fine. Might be worth mentioning in the documentation that using docker snap doesn't work.

Good idea. I'll add the incompatibility about Snap. Or at least the non-out-of-boxness
